### PR TITLE
build: handle push/load shorthands for multi exporters

### PR DESCRIPTION
**Pull request by crazy-max**
fixes https://github.com/docker/build-push-action/issues/727

We forgot to handle `push` and `load` shorthands in https://github.com/docker/buildx/pull/2290.

~Keeping this one in draft while I'm working on integration tests.~

~EDIT: Argh GitHub is broken https://www.githubstatus.com/incidents/wcl1sw4mzg60~

### CODE REVIEW

File: commands/bake.go
Comment (crazy-max) [inline commands/bake.go:1]: Adding integration tests for bake as well. Wants to make sure override behavior is as expected.

File: controller/build/build.go [hunk 1]
```diff
@@ -99,38 +99,18 @@ func RunBuild(ctx context.Context, dockerCli command.Cli, in controllerapi.Build
...
-			}}
-		} else {
-			switch outputs[0].Type {
-			case "image":
-				outputs[0].Attrs["push"] = "true"
-			default:
-				return nil, nil, errors.Errorf("push and %q output can't be used together", outputs[0].Type)
-			}
-		}
+		outputs = append(outputs, client.ExportEntry{
```
Comment (tonistiigi) [inline controller/build/build.go:102]: I think we still need to do specific behavior based on the current value. If the existing value has image exports then these should be set to be pushed.

Additionally, if we don't do this then it is now an error for older daemons that just defined the same output multiple times.

File: controller/build/build.go [hunk 2]
```diff
@@ -99,38 +99,18 @@ func RunBuild(ctx context.Context, dockerCli command.Cli, in controllerapi.Build
...
-				Attrs: map[string]string{},
-			}}
-		} else {
-			switch outputs[0].Type {
-			case "docker":
-			default:
-				return nil, nil, errors.Errorf("load and %q output can't be used together", outputs[0].Type)
-			}
-		}
+		outputs = append(outputs, client.ExportEntry{
```
Comment (tonistiigi) [inline controller/build/build.go:110]: Similarly, if docker output already exists then this should not do anything.

Comment (crazy-max) [reply controller/build/build.go]: This is a bit more than that I think. `docker` exporter can also output to a file so we need to keep this case but also be able to output to docker store if `--load`:

```
$ docker buildx build --output type=docker --output type=docker,dest=build.tar
```

File: controller/build/build.go [hunk 3]
```diff
@@ -99,36 +99,52 @@ func RunBuild(ctx context.Context, dockerCli command.Cli, in controllerapi.Build
 		return nil, nil, err
 	}
 	if in.ExportPush {
-		if in.ExportLoad {
-			return nil, nil, errors.Errorf("push and load may not be set together at the moment")
-		}
 		if len(outputs) == 0 {
```
Comment (tonistiigi) [inline controller/build/build.go:102]: This case is not needed anymore. Else case behaves same way if `len(outputs) == 0`. Same for `docker`.

Comment (crazy-max) [reply controller/build/build.go:102]: Ah right, we are already checking if push/load is already used.

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**